src/Header.cpp

```diff
@@ -0,0 +1,37 @@
+#include "lift/Header.hpp"
+
+#include <string>
+
+namespace lift {
+
+Header::Header(
+    std::string_view name,
+    std::string_view value)
+{
+    m_header.reserve(name.length() + value.length() + 2);
+    m_header.append(name);
+    m_header.append(": ");
+    m_header.append(value);
+
+    auto colon_pos = m_header.find(':');
+    if (colon_pos != std::string_view::npos) {
```

Comment:
I think this can definitely be rewritten so we don't need a check for something we know can't happen